docs/resources/azure_network_interface.md

```diff
@@ -0,0 +1,101 @@
+
+## Availability
+
+### Installation
+
+This resource is available in the [InSpec Azure resource pack](https://github.com/inspec/inspec-azure). 
+For an example `inspec.yml` file and how to set up your Azure credentials, refer to resource pack [README](../../README.md#Service-Principal).
+
+## Syntax
+
+An `azure_network_interface` resource block identifies an AKS Cluster by `name` and `resource_group` or the `resource_id`.
+```ruby
+describe azure_network_interface(resource_group: 'example', name: 'networkInterfaceName') do
+  it { should exist }
+end
+```
+```ruby
+describe azure_network_interface(resource_id: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/networkInterfaces/{networkInterfaceName}') do
+  it { should exist }
+end
+```
+## Parameters
+
+| Name                           | Description                                                                       |
+|--------------------------------|-----------------------------------------------------------------------------------|
+| resource_group                 | Azure resource group that the targeted resource resides in. `MyResourceGroup`     |
+| name                           | Name of the AKS cluster to test. `networkInterfaceName`                                      |
+| resource_id                    | The unique resource ID. `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/networkInterfaces/{networkInterfaceName}` |
+
+Either one of the parameter sets can be provided for a valid query:
+- `resource_id`
+- `resource_group` and `name`
+
+## Properties
+
+| Property                | Description |
+|-------------------------|-------------|
+| primary?                | Indicates whether this is a primary network interface on a virtual machine. |
+| ip_configurations       | A list of [IPConfigurations](https://docs.microsoft.com/en-us/rest/api/virtualnetwork/networkinterfaceipconfigurations/get#networkinterfaceipconfiguration) of the network interface. |
+| private_ip              | The private IP address of the interrogated network interface's first IP configuration. |
+| has_private_address_ip? | Indicates whether the interrogated network interface's first IP configuration has a private IP address. |
+| public_ip               | The public IP address of the interrogated network interface's first IP configuration. |
+| has_public_address_ip?  | Indicates whether the interrogated network interface's first IP configuration has a public IP address. |
+For properties applicable to all resources, such as `type`, `name`, `id`, `properties`, refer to [`azure_generic_resource`](azure_generic_resource.md#properties).
+
+Also, refer to [Azure documentation](https://docs.microsoft.com/en-us/rest/api/virtualnetwork/networkinterfaces/get#networkinterface) for other properties available. 
+Any attribute in the response may be accessed with the key names separated by dots (`.`).
```

Comment:
I wonder if we should add an example to the end of this statement
```
Any attribute in the response may be accessed with the key names separated by dots (`.`). eg. `properties.<attribute>`
```

Reply:
Good catch, I'll update it.
I try to give such an example in the examples section usually.

Reply:
Updated.